lib/pacemaker/pcmk_output.c

```diff
@@ -647,47 +647,91 @@ health_xml(pcmk__output_t *out, va_list args)
     return pcmk_rc_ok;
 }
 
-PCMK__OUTPUT_ARGS("pacemakerd-health", "const char *", "const char *", "const char *")
+PCMK__OUTPUT_ARGS("pacemakerd-health", "const char *", "int", "const char *",
+                  "const char *")
 static int
 pacemakerd_health(pcmk__output_t *out, va_list args)
 {
     const char *sys_from = va_arg(args, const char *);
-    const char *state = va_arg(args, const char *);
+    enum pcmk_pacemakerd_state state =
+        (enum pcmk_pacemakerd_state) va_arg(args, int);
+    const char *state_s = va_arg(args, const char *);
     const char *last_updated = va_arg(args, const char *);
 
+    if (state_s == NULL) {
+        state_s = pcmk__pcmkd_state_enum2friendly(state);
+    }
     return out->info(out, "Status of %s: '%s' (last updated %s)",
-                     pcmk__s(sys_from, "unknown node"),
-                     pcmk__s(state, "unknown state"),
+                     pcmk__s(sys_from, "unknown node"), state_s,
```

Comment:
I kept `"unknown node"` as the `sys_from` default in case it ever can come from a node. As far as I've seen, `sys_from` in this context is always `CRM_SYSTEM_MCP` (`"pacemakerd"`). It might be more accurate to say something like "unknown system", or even to hard-code `CRM_SYSTEM_MCP` if there's no other possibility. (Maybe `"pacemaker-remote"` or similar is a possibility?)